File: SMedia/SMedia/Models/Post.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SMedia.Models
{
    public class CreationPost
    {
        public string Content { get; set; }
        public long AuthorId { get; set; }
        public long CommunityId { get; set; }
        public long? lastPostId { get; set; }
    }
    public class Post
    {
        public long Id { get; set; }
        public string Content { get; set; }
        public DateTime CreationDate { get; set; }
        public long AuthorId { get; set; }
        public long CommunityId { get; set; }
        public long? LastPostId { get; set; }
        public bool Active { get; set; }
        
        public virtual User Author { get; set; }
        public virtual Community Community { get; set; }
        public virtual Post LastPost { get; set; }
        public virtual Post BeforePost { get; set; }


        public virtual ICollection<FavoritePost> FavoritePost { get; set; }
        public virtual ICollection<PostPicture> PostPicture { get; set; }
        public virtual ICollection<Reaction> Reaction { get; set; }
        public virtual ICollection<Viewed> Viewed { get; set; }
    }
}

File: woofer-frontend/src/models/post.js
export class Post{

    constructor(id = 0, content = '', creationDate = '', author = {}, community = {}) {

        this.id = id;
        this.content = content;
        this.creationDate = creationDate;
        this.author = author;
        this.community = community;

    }

    id;
    content;
    creationDate;
    author; // expects User from src/models/user
    community; // expects CommunityView from src/models/community

}
